### DISCUSSION TITLE
FCReplaceAll not replacing four vector values

### DISCUSSION THREAD

**Discussion by Pholipo** [FeynCalc/feyncalc#276] [Q&A] [answered]
Greetings,

Why isn't the ReplaceAll working with the four-vector below? I actually want to make more replacements in the output variable ex5. I basically need to implement a shift in the k 4-vector in the whole expression resulting from these common shifts made in Feynman Parametrization. The D there is the dimension made variable in Dimensional Regularization.
![image](https://github.com/FeynCalc/feyncalc/assets/97998280/354b5c8f-e35b-4f43-8f3f-3b23192607c8)

**Comment by vsht** [comment, marked as answer]
Hi,

it's because you're ignoring the difference between internal and 
external representations

https://feyncalc.github.io/FeynCalcBookDev/InternalExternal.html

I suggest that you go through the tutorial in the manual first before
trying to calculate something. Those are very basic things and without
having a good understanding how they work you won't get far.



Am 25.06.24 um 01:48 schrieb Pholipo:

**Reply by Pholipo** [reply]
Thanks a lot!
